code/modules/admin/verbs/buildmode.dm

```diff
@@ -354,10 +361,17 @@
 						if(GLOB.reloading_map)
 							to_chat(user, "<span class='boldwarning'>You are already reloading an area! Please wait for it to fully finish loading before trying to load another!</span>")
 							return
-					G.defineRegion(cornerA,cornerB,1)
-					G.generate()
+					for(var/S in block(cornerA, cornerB))
```

Comment:
revert your last commit, and just do:

```
G.defineRegion(cornerA, cornerB, 1)
var/list/preview = list()
for(var/t in G.map)
    preview += image('icons/turf/overlays.dmi', t ,"greenOverlay")
user.client.images += preview
var/confirm = alert("Are you sure you want run the map generator?", "Run generator", "Yes", "No")
if(confirm == "Yes")
    G.generate()
user.client.images -= preview
```

Putting ```defineRegion()``` back where it was (before the alert) means ```block()``` only has to be called once (inside of ```defineRegion()```) to get all the turfs

Reply:
ah, thanks - I didn't know if there was a way to access the turfs - probably should have looked harder

Reply:
I didn't write a huge readme for nothing!